src/jit/gentree.cpp

```diff
@@ -16563,6 +16563,107 @@ CORINFO_CLASS_HANDLE Compiler::gtGetClassHandle(GenTree* tree, bool* isExact, bo
                     objClass = sig.retTypeClass;
                 }
             }
+            else if (call->gtCallType == CT_HELPER)
+            {
+                const CorInfoHelpFunc helper       = eeGetHelperNum(call->gtCallMethHnd);
+                bool                  isCastHelper = false;
+
+                switch (helper)
+                {
+                    case CORINFO_HELP_TYPEHANDLE_TO_RUNTIMETYPE:
+                    case CORINFO_HELP_TYPEHANDLE_TO_RUNTIMETYPE_MAYBENULL:
+                    {
+                        // Note for some runtimes these helpers return exact types.
+                        //
+                        // But in those cases the types are also sealed, so there's no
+                        // need to claim exactness here.
+                        const bool           helperResultNonNull = (helper == CORINFO_HELP_TYPEHANDLE_TO_RUNTIMETYPE);
+                        CORINFO_CLASS_HANDLE runtimeType = info.compCompHnd->getBuiltinClass(CLASSID_RUNTIME_TYPE);
+
+                        objClass   = runtimeType;
+                        *isNonNull = helperResultNonNull;
+                        break;
+                    }
+
+                    case CORINFO_HELP_CHKCASTCLASS:
+                    case CORINFO_HELP_CHKCASTANY:
+                    case CORINFO_HELP_CHKCASTARRAY:
+                    case CORINFO_HELP_CHKCASTINTERFACE:
+                    case CORINFO_HELP_CHKCASTCLASS_SPECIAL:
+                    case CORINFO_HELP_ISINSTANCEOFINTERFACE:
+                    case CORINFO_HELP_ISINSTANCEOFARRAY:
+                    case CORINFO_HELP_ISINSTANCEOFCLASS:
+                    case CORINFO_HELP_ISINSTANCEOFANY:
+                    {
+                        // Fetch the class handle from the helper call arglist
+                        GenTreeArgList*      args    = call->gtCallArgs;
+                        GenTree*             typeArg = args->Current();
+                        CORINFO_CLASS_HANDLE castHnd = gtGetHelperArgClassHandle(typeArg);
+
+                        // We generally assume the type being cast to the best type
```

Comment:
missing "is"
he type being cast to **is** the best

Reply:
Thanks, will fix.